app/main.py

```diff
@@ -1,6 +1,18 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    memory = {}
+
+    def wrapper(*args, **kwargs) -> Any:
```

Comment:
You could use a decorator from `functools` to preserve the wrapped function name and docstring.